lib/puppet/provider/vc_dvswitch/default.rb

```diff
@@ -18,6 +18,10 @@ def create
     create_spec = RbVmomi::VIM::DVSCreateSpec.new
     create_spec.configSpec = RbVmomi::VIM::DVSConfigSpec.new
     create_spec.configSpec.name = basename
+    if @dvswitch.nil? && resource[:vds_version]
```

Comment:
@dvswitch.nil? seems unnecessarily redundant since this is the create method and @dvswitch is set by the method triggered by the exists? method.